src/client/eppo-client.spec.ts

```diff
@@ -208,6 +214,158 @@ describe('EppoClient E2E test', () => {
+          'alice',
+          subjectAttributes,
+          0,
+        );
+        const expected: AssignmentDetails<number> = {
+          value: 2,
+          variationKey: 'two',
+          variationValue: 2,
+          flagEvaluationCode: 'MATCH',
+          flagEvaluationDescription:
+            'alice belongs to the range of traffic assigned to "two" defined in allocation "50/50 split".',
+          matchedRule: null,
+          matchedAllocation: {
+            key: '50/50 split',
+            allocationEvaluationCode: AllocationEvaluationCode.MATCH,
+            orderPosition: 1,
+          },
+          unmatchedAllocations: [
+            {
+              key: 'targeted allocation',
+              allocationEvaluationCode: AllocationEvaluationCode.FAILING_RULE,
+              orderPosition: 0,
+            },
+          ],
+          unevaluatedAllocations: [],
+        };
+        expect(result).toEqual(expected);
+      });
+
+      it('should handle matching a split allocation with a matched rule', () => {
+        const client = new EppoClient(storage);
+        client.setIsGracefulFailureMode(false);
+        const subjectAttributes = { id: 'alice', email: 'alice@external.com', country: 'Brazil' };
+        const result = client.getStringAssignmentDetails(
+          'new-user-onboarding',
+          'alice',
+          subjectAttributes,
+          '',
+        );
+        const expected: AssignmentDetails<string> = {
+          value: 'control',
+          flagEvaluationCode: 'MATCH',
+          flagEvaluationDescription:
+            'Supplied attributes match rules defined in allocation "experiment" and alice belongs to the range of traffic assigned to "control".',
+          variationKey: 'control',
+          variationValue: 'control',
+          matchedRule: {
+            conditions: [
+              {
+                attribute: 'country',
+                operator: OperatorType.NOT_ONE_OF,
+                value: ['US', 'Canada', 'Mexico'],
+              },
+            ],
+          },
+          matchedAllocation: {
+            key: 'experiment',
+            allocationEvaluationCode: AllocationEvaluationCode.MATCH,
+            orderPosition: 2,
+          },
+          unmatchedAllocations: [
+            {
+              key: 'id rule',
+              allocationEvaluationCode: AllocationEvaluationCode.FAILING_RULE,
+              orderPosition: 0,
+            },
+            {
+              key: 'internal users',
+              allocationEvaluationCode: AllocationEvaluationCode.FAILING_RULE,
+              orderPosition: 1,
+            },
+          ],
+          unevaluatedAllocations: [
+            {
+              key: 'rollout',
+              allocationEvaluationCode: AllocationEvaluationCode.UNEVALUATED,
+              orderPosition: 3,
+            },
+          ],
+        };
+        expect(result).toEqual(expected);
+      });
+
+      it('should handle unrecognized flags', () => {
```

Comment:
I prob need to add a few more tests here

Reply:
An error case would be useful too (you'd have to stub out the evaluator to throw an exception or something)